### PR TITLE
Update layout - application pages (unsubmitted & review)

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

The application pages have been given a slight overhaul in terms of layout.

See [layout / copy changes proposed Menu structure](https://docs.google.com/document/d/1Zm9xxn9w9iEMNvCXq-O7yxOgG34d1oRNiw3Av47zyIM/edit?usp=sharing).

## Changes proposed in this pull request

- This PR updates the layout only.
- Another PR will follow to update routes and links (e.g. so that they match changes to any section items)

## Guidance to review

- Probably the easiest way to review this will be to spin up the branch locally (or go to the review app) and [compare with the prototype](https://apply-beta-prototype.herokuapp.com/) for both submitted and pending applications

## Link to Trello card

https://trello.com/c/KYTscZna/2511-%F0%9F%8F%88-dev-update-menu-structure-linked-page-headings-and-other-flagged-content

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -31,7 +31,8 @@ en:
     forbidden: Access denied
     internal_server_error: Sorry, there is a problem with the service
     unprocessable_entity: Sorry, there is a problem with the service
-    personal_details: Personal details
+#    personal_details: Personal details
```
Comment (davidgisbey) [inline config/locales/en.yml]: I think you misssed this.

**Review by frankieroberto — APPROVED** [review]
Looks good to me. 👍

**Review by davidgisbey — APPROVED** [review]
Sorry, i forgot to approve. It looks good 👍 1

File: app/views/candidate_interface/application_form/_review.html.erb [hunk 2]
```diff
@@ -1,91 +1,109 @@
...
+
+  <h3 class="govuk-heading-m"><%= t('page_titles.personal_information') %></h3>
+  <%= render(CandidateInterface::PersonalDetailsReviewComponent.new(application_form: application_form, editable: editable, missing_error: missing_error)) %>
+
+  <h3 class="govuk-heading-m"><%= t('page_titles.contact_details') %></h3>
+  <%= render(CandidateInterface::ContactDetailsReviewComponent.new(application_form: application_form, editable: editable, missing_error: missing_error, submitting_application: true)) %>
+</section>
+
+<section class="govuk-!-margin-bottom-8">
+  <h2 class="govuk-heading-m govuk-!-font-size-27">Qualifications</h2>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/application_form/_review.html.erb:55]: Qualification sections should be ordered thus (alphabetical except other section, which comes last):

* Degree
* English GCSE or equivalent
* English as a foreign language assessment
* Maths GCSE or equivalent
* Science GCSE or equivalent
* Academic and other relevant qualifications

File: app/components/candidate_interface/application_form_course_choices_component.html.erb [hunk 3]
```diff
@@ -1,19 +1,21 @@
-<h2 class="govuk-heading-m govuk-!-font-size-27 govuk-!-margin-bottom-2">
-  <%= page_title %>
-</h2>
+<section class="govuk-!-margin-bottom-8">
+  <h2 class="govuk-heading-m govuk-!-font-size-27">
+    <%= page_title %>
+  </h2>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/application_form_course_choices_component.html.erb:4]: Given how we’re indenting headings elsewhere on this page, and that this is a short line of code:

```suggestion
  <h2 class="govuk-heading-m govuk-!-font-size-27"><%= page_title %></h2>
```

File: app/views/candidate_interface/application_form/_review.html.erb [hunk 4]
```diff
@@ -1,91 +1,109 @@
...
+  <% if !CandidateInterface::EndOfCyclePolicy.can_add_course_choice?(@application_form)%>
+    <h2 class="govuk-heading-m govuk-!-font-size-27 govuk-!-margin-bottom-2"><%= t('page_titles.course_choices') %></h2>
+    <p class="govuk-body">
+      You’ll be able to find courses in <%= days_until_find_reopens %> <%= 'day'.pluralize(days_until_find_reopens) %>
+      (<%= EndOfCycleTimetable.find_reopens.to_s(:govuk_date) %>).
+      You can keep making changes to the rest of your application until then.
+    </p>
   <% else %>
-    <%= render(CandidateInterface::SectionMissingBannerComponent.new(section: 'efl', section_path: candidate_interface_english_foreign_language_start_path, error: missing_error)) %>
+    <h2 class="govuk-heading-m govuk-!-font-size-27">Courses</h2>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/application_form/_review.html.erb:13]: * Use same localised value across all application pages that use this heading.
* The courses heading should only be pluralised on initial application; apply again heading should not be pluralised as you can only have one course choice.

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Couple of things to address:

* Ordering of qualification sections
* Pluralisation of ‘Course’ and ‘Courses’ based on application stage
* Grouping of heading localisations (move out from under `page_titles`), and using these for all headings across all relevant pages
* Indentation of component parameters

File: app/components/candidate_interface/application_form_course_choices_component.html.erb [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
+  <ol class="app-task-list">
+    <li class="app-task-list__item">
+      <%= render(
+            TaskListItemComponent.new(
+              text: text,
+              completed: completed?,
+              path: view_courses_path,
+              )
+          ) %>
+    </li>
```
[the remark this answers is not in the document]
Comment (TobyRet) [reply app/components/candidate_interface/application_form_course_choices_component.html.erb:19]: 👍

File: app/views/candidate_interface/application_form/_review.html.erb [hunk 6]
```diff
@@ -1,91 +1,109 @@
...
+    <%= render(
+          CandidateInterface::CourseChoicesReviewComponent.new(
+            application_form: application_form,
+            editable: editable,
+            heading_level: 3,
+            show_incomplete: true,
+            missing_error: missing_error,
+            application_choice_error: application_choice_error,
+            )
+        ) %>
```
[the remark this answers is not in the document]
Comment (TobyRet) [reply app/views/candidate_interface/application_form/_review.html.erb:23]: 👍

File: app/views/candidate_interface/application_form/_review.html.erb [hunk 7]
```diff
@@ -1,91 +1,109 @@
...
+            error: missing_error
+          ),
+          ) %>
+  <% else %>
+    <%= render(CandidateInterface::ReferencesReviewComponent.new(references: application_form.application_references.includes(:application_form).feedback_provided, editable: false)) %>
+  <% end %>
+</section>
+
+<section class="govuk-!-margin-bottom-8">
+  <h2 class="govuk-heading-m govuk-!-font-size-27"><%= t('page_titles.personal_details') %></h2>
```
[the remark this answers is not in the document]
Comment (TobyRet) [reply app/views/candidate_interface/application_form/_review.html.erb:45]: 👍

File: app/views/candidate_interface/application_form/_review.html.erb [hunk 8]
```diff
@@ -1,91 +1,109 @@
...
+  <%= render(CandidateInterface::OtherQualificationsReviewComponent.new(application_form: application_form, editable: editable, heading_level: 4, missing_error: missing_error, submitting_application: true)) %>
+
+  <% if @application_form.efl_section_required? %>
+    <h3 class="govuk-heading-m">English as a foreign language assessment</h3>
+    <% if @application_form.english_proficiency.present? %>
+      <%= render(CandidateInterface::ChooseEflReviewComponent.call(@application_form.english_proficiency)) %>
+    <% else %>
+      <%= render(CandidateInterface::SectionMissingBannerComponent.new(section: 'efl', section_path: candidate_interface_english_foreign_language_start_path, error: missing_error)) %>
+    <% end %>
+  <% end %>
```
[the remark this answers is not in the document]
Comment (TobyRet) [reply app/views/candidate_interface/application_form/_review.html.erb:81]: 👍

File: app/views/candidate_interface/application_form/_review.html.erb (hunk 2 above)
Comment (TobyRet) [reply app/views/candidate_interface/application_form/_review.html.erb:55]: 👍

File: app/components/candidate_interface/application_form_course_choices_component.html.erb (hunk 3 above)
Comment (TobyRet) [reply app/components/candidate_interface/application_form_course_choices_component.html.erb:4]: 👍

File: app/views/candidate_interface/application_form/_review.html.erb [hunk 9]
```diff
@@ -1,91 +1,109 @@
...
-<h3 class="govuk-heading-m"><%= t('page_titles.other_qualification') %></h3>
-<%= render(CandidateInterface::OtherQualificationsReviewComponent.new(application_form: application_form, editable: editable, heading_level: 4, missing_error: missing_error, submitting_application: true)) %>
-
-<% if @application_form.efl_section_required? %>
-  <h3 class="govuk-heading-m">English as a foreign language assessment</h3>
-  <% if @application_form.english_proficiency.present? %>
-    <%= render(CandidateInterface::ChooseEflReviewComponent.call(@application_form.english_proficiency)) %>
+<section class="govuk-!-margin-bottom-8">
+  <% if !CandidateInterface::EndOfCyclePolicy.can_add_course_choice?(@application_form)%>
+    <h2 class="govuk-heading-m govuk-!-font-size-27 govuk-!-margin-bottom-2"><%= t('page_titles.course_choices') %></h2>
```
[the remark this answers is not in the document]
Comment (TobyRet) [reply app/views/candidate_interface/application_form/_review.html.erb:6]: Yep makes sense 👍

**Comment by paulrobertlloyd** [comment]
@TobyRet Sorry, I started looking a little bit too closely… and made some further changes:

* Ensured all section group headings are in that localisation, and all section items are localised (i.e. reference links)
* Fixed a missing `<%= %>` around a localised heading on the application review page
* Fixed a stray ‘unsexy’ quote for training with a disability section
* Corrected the markup around the course choice section in its different states
* Removed localisations in for course choice component controllers, used updated localised strings in component instead
* Fixed markup around the page titles and supporting hint text on ‘Your application’ and ‘Application dashboard’ pages
* Updated the indentation of sections on application index so that page structure is clearer

I think this covers everything. There are however a few things for a later PR…

* We use the ‘Apply again’ course component, and this uses non-pluralised ‘course’ labels. However, applying again does not always mean you have 1 choice – you can apply again across cycles and have 3 choices again. I wonder if we should merge `ApplicationFormApplyAgainCourseChoiceComponent` and `ApplicationFormCourseChoicesComponent` (they appear near identical) and pluralise text based on the number of available choices, not based on whether `candidate_has_previously_applied`? Will create a Trello card.

* Also… will check with content team, if active titles are preferred, we should use ‘Choose your course[s]’ for the section page title as well the link to it (and then like other sections, we can use the page title on the index too).

**Comment by TobyRet** [comment]
> @TobyRet Sorry, I started looking a little bit too closely… and made some further changes:
> 
> * Ensured all section group headings are in that localisation, and all section items are localised (i.e. reference links)
> * Fixed a missing `<%= %>` around a localised heading on the application review page
> * Fixed a stray ‘unsexy’ quote for training with a disability section
> * Corrected the markup around the course choice section in its different states
> * Removed localisations in the 2 course choice component controllers, now using updated localised strings in component instead
> * Fixed markup around the page titles and supporting hint text on ‘Your application’ and ‘Application dashboard’ pages
> * Updated the indentation of sections on application index so that page structure is clearer
> 
> I think this covers everything. There are however a few things for a later PR…
> 
> * We use the ‘Apply again’ course component, and this uses non-pluralised ‘course’ labels. However, applying again does not always mean you have 1 choice – you can apply again across cycles and have 3 choices again. I wonder if we should merge `ApplicationFormApplyAgainCourseChoiceComponent` and `ApplicationFormCourseChoicesComponent` (they appear near identical) and pluralise text based on the number of available choices, not based on whether `candidate_has_previously_applied`? [Trello card](https://trello.com/c/YeVTUvOu/)
> * Also… will check with content team, if active titles are preferred, we should use ‘Choose your course[s]’ for the section page title as well the link to it (and then like other sections, we can use the page title on the index too). [Trello card](https://trello.com/c/v9auchn6/)

Thanks for taking a look @paulrobertlloyd 

Good to have a second pair of eyes on this, given how many changes have been clocked up already.

The above makes sense to me.

Another observation is that the change from 'contact details' to 'contact information' has resulted in a ripple-effect of changes right throughout the codebase 😢. A lot of the changes have been addressed at the presentation layer, but we still need to apply the changes at the model / db layer. 

This PR also does not address updating links and routes as a result of the copy changes in the various sections. I was planning to do this in a follow up PR once the team are happy with the proposed changes so far. We will also need to add a number of redirects from old the old paths to the new paths.

**Review by paulrobertlloyd — APPROVED** [review]

**Pull request merged**